File: nan21.dnet.module.ad/nan21.dnet.module.ad.presenter/src/main/java/net/nan21/dnet/module/ad/usr/ds/model/MenuItemRtLovDs.java
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */
package net.nan21.dnet.module.ad.usr.ds.model;

import net.nan21.dnet.core.api.annotation.SortField;
import net.nan21.dnet.core.presenter.model.base.AbstractTypeDs;

import net.nan21.dnet.module.ad.usr.domain.entity.MenuItem;
import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;

@Ds(entity = MenuItem.class, jpqlWhere = " e.active = true ", sort = { @SortField(field = MenuItemRtLovDs.fSEQUENCENO) })
public class MenuItemRtLovDs extends AbstractTypeDs<MenuItem> {

    public static final String fSEQUENCENO = "sequenceNo";
    public static final String fMENUITEMID = "menuItemId";
    public static final String fMENUITEM = "menuItem";
    public static final String fMENUID = "menuId";
    public static final String fMENU = "menu";
    public static final String fTEXT = "text";
    public static final String fFRAME = "frame";
    public static final String fBUNDLE = "bundle";
    public static final String fLEAF = "leaf";

    @DsField()
    private Integer sequenceNo;

    @DsField(join = "left", path = "menuItem.id")
    private Long menuItemId;

    @DsField(join = "left", path = "menuItem.name")
    private String menuItem;

    @DsField(join = "left", path = "menu.id")
    private Long menuId;

    @DsField(join = "left", path = "menu.name")
    private String menu;

    @DsField(path = "title")
    private String text;

    @DsField()
    private String frame;

    @DsField()
    private String bundle;

    @DsField(fetch = false, path = "leafNode")
    private Boolean leaf;

    public MenuItemRtLovDs() {
        super();
    }

    public MenuItemRtLovDs(MenuItem e) {
        super(e);
    }

    public Integer getSequenceNo() {
        return this.sequenceNo;
    }

    public void setSequenceNo(Integer sequenceNo) {
        this.sequenceNo = sequenceNo;
    }

    public Long getMenuItemId() {
        return this.menuItemId;
    }

    public void setMenuItemId(Long menuItemId) {
        this.menuItemId = menuItemId;
    }

    public String getMenuItem() {
        return this.menuItem;
    }

    public void setMenuItem(String menuItem) {
        this.menuItem = menuItem;
    }

    public Long getMenuId() {
        return this.menuId;
    }

    public void setMenuId(Long menuId) {
        this.menuId = menuId;
    }

    public String getMenu() {
        return this.menu;
    }

    public void setMenu(String menu) {
        this.menu = menu;
    }

    public String getText() {
        return this.text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public String getFrame() {
        return this.frame;
    }

    public void setFrame(String frame) {
        this.frame = frame;
    }

    public String getBundle() {
        return this.bundle;
    }

    public void setBundle(String bundle) {
        this.bundle = bundle;
    }

    public Boolean getLeaf() {
        return this.leaf;
    }

    public void setLeaf(Boolean leaf) {
        this.leaf = leaf;
    }

}

File: nan21.dnet.module.ad/nan21.dnet.module.ad.ui.extjs/src/main/webapp/src/ds/MenuItemRtLovDs.js
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */   	 
Ext.define("net.nan21.dnet.module.ad.usr.ds.model.MenuItemRtLovDs" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"name", type:"string"},
		{name:"active", type:"boolean"},
		{name:"description", type:"string"},
		{name:"id", type:"int", useNull:true},
		{name:"uuid", type:"string"},
		{name:"clientId", type:"int", useNull:true},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"modifiedBy", type:"string"},
		{name:"version", type:"int", useNull:true},
		{name:"entityFQN", type:"string"},
		{name:"sequenceNo", type:"int", useNull:true},
		{name:"menuItemId", type:"int", useNull:true},
		{name:"menuItem", type:"string"},
		{name:"menuId", type:"int", useNull:true},
		{name:"menu", type:"string"},
		{name:"text", type:"string"},
		{name:"frame", type:"string"},
		{name:"bundle", type:"string"},
		{name:"leaf", type:"boolean"}
	],
	validations: [  
		{field: "name", type: 'presence'}
	]	
	
});

Ext.define("net.nan21.dnet.module.ad.usr.ds.model.MenuItemRtLovDsFilter" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"name", type:"string"},
		{name:"active", type:"boolean", useNull:true},
		{name:"description", type:"string"},
		{name:"id", type:"int", useNull:true},
		{name:"uuid", type:"string"},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"modifiedBy", type:"string"},
		{name:"entityFQN", type:"string"},
		{name:"sequenceNo", type:"int", useNull:true},
		{name:"sequenceNo_From",type:"int", useNull:true},
		{name:"sequenceNo_To",type:"int", useNull:true},
		{name:"menuItemId", type:"int", useNull:true},
		{name:"menuItemId_From",type:"int", useNull:true},
		{name:"menuItemId_To",type:"int", useNull:true},
		{name:"menuItem", type:"string"},
		{name:"menuId", type:"int", useNull:true},
		{name:"menuId_From",type:"int", useNull:true},
		{name:"menuId_To",type:"int", useNull:true},
		{name:"menu", type:"string"},
		{name:"text", type:"string"},
		{name:"frame", type:"string"},
		{name:"bundle", type:"string"},
		{name:"leaf", type:"boolean", useNull:true}
	] 
});
	
Ext.define("net.nan21.dnet.module.ad.usr.ds.param.MenuItemRtLovDsParam", {
	extend: 'Ext.data.Model',
	fields: [ 
	]
}); 
 